akamai/cache-client_test.go

```diff
@@ -129,19 +143,54 @@ func (as *akamaiServer) akamaiHandler(w http.ResponseWriter, r *http.Request) {
 	as.sendResponse(w, resp)
 }
 
+func (as *akamaiServer) checkSignature(r *http.Request, body []byte) error {
+	bodyHash := sha256.Sum256(body)
+	bodyHashB64 := base64.StdEncoding.EncodeToString(bodyHash[:])
+
+	authorization := r.Header.Get("Authorization")
+	authValues := make(map[string]string)
+	for _, v := range strings.Split(authorization, ";") {
+		splitValue := strings.Split(v, "=")
+		authValues[splitValue[0]] = splitValue[1]
+	}
+	headerTimestamp := authValues["timestamp"]
+	shortenedHeader := strings.Split(authorization, "signature=")[0]
+	// The signature is just long enough to get b64 padding which gets messed up
```

Comment:
Seems like using `SplitN` here would solve this problem? Just split on only the first `=` then you don't have to use another method to split out the signature at the end.